lighthouse-core/gather/driver/navigation.js

```diff
@@ -94,9 +96,14 @@ async function gotoURL(driver, url, options) {
   await session.sendCommand('Page.enable');
   await session.sendCommand('Page.setLifecycleEventsEnabled', {enabled: true});
 
-  // No timeout needed for Page.navigate. See https://github.com/GoogleChrome/lighthouse/pull/6413
-  session.setNextProtocolTimeout(Infinity);
-  const waitforPageNavigateCmd = session.sendCommand('Page.navigate', {url});
+  let waitForPageNavigateCmd;
+  if (typeof requestor === 'string') {
+    // No timeout needed for Page.navigate. See https://github.com/GoogleChrome/lighthouse/pull/6413
+    session.setNextProtocolTimeout(Infinity);
+    waitForPageNavigateCmd = session.sendCommand('Page.navigate', {url: requestor});
+  } else {
+    waitForPageNavigateCmd = requestor();
```

Comment:
It could be useful to make a new status logging step here, and keep the original for this method the same.

Reply:
We will have to replace the URL in the original log message with something for user controlled navigations. I think that would make a log statement here redundant.